src/control/XournalMain.cpp

```diff
@@ -276,238 +297,73 @@ auto XournalMain::exportPdf(const char* input, const char* output) -> int {
+struct XournalMainPrivate {
+    XournalMainPrivate() = default;
+    XournalMainPrivate(XournalMainPrivate&&) = delete;
+    XournalMainPrivate(XournalMainPrivate const&) = delete;
+    XournalMainPrivate operator=(XournalMainPrivate&&) = delete;
+    XournalMainPrivate operator=(XournalMainPrivate const&) = delete;
 
-    GError* error = nullptr;
-    GOptionContext* context = g_option_context_new("FILE");
+    ~XournalMainPrivate() {
+        g_strfreev(optFilename);
+        g_free(pdfFilename);
+        g_free(imgFilename);
+    }
 
-    gchar** optFilename = nullptr;
-    gchar* pdfFilename = nullptr;
-    gchar* imgFilename = nullptr;
+    gchar** optFilename{};
+    gchar* pdfFilename{};
+    gchar* imgFilename{};
     gboolean showVersion = false;
     int openAtPageNumber = -1;
-
-    string create_pdf = _("PDF output filename");
-    string create_img = _("Image output filename (.png / .svg)");
-    string page_jump = _("Jump to Page (first Page: 1)");
-    string audio_folder = _("Absolute path for the audio files playback");
-    string version = _("Get version of xournalpp");
-    GOptionEntry options[] = {{"create-pdf", 'p', 0, G_OPTION_ARG_FILENAME, &pdfFilename, create_pdf.c_str(), nullptr},
-                              {"create-img", 'i', 0, G_OPTION_ARG_FILENAME, &imgFilename, create_img.c_str(), nullptr},
-                              {"page", 'n', 0, G_OPTION_ARG_INT, &openAtPageNumber, page_jump.c_str(), "N"},
-                              {G_OPTION_REMAINING, 0, 0, G_OPTION_ARG_FILENAME_ARRAY, &optFilename, "<input>", nullptr},
-                              {"version", 0, 0, G_OPTION_ARG_NONE, &showVersion, version.c_str(), nullptr},
-                              {nullptr}};
-
-    g_option_context_add_main_entries(context, options, GETTEXT_PACKAGE);
-    // parse options, so we don't need gtk_init, but don't init display (so we have a commandline mode)
-    g_option_context_add_group(context, gtk_get_option_group(false));
-    if (!g_option_context_parse(context, &argc, &argv, &error)) {
-        g_error("%s", error->message);
-        g_error_free(error);
-        gchar* help = g_option_context_get_help(context, true, nullptr);
-        g_message("%s", help);
-        g_free(help);
-        error = nullptr;
-    }
-    g_option_context_free(context);
-
-    if (pdfFilename && optFilename && *optFilename) {
-        return exportPdf(*optFilename, pdfFilename);
-    }
-    if (imgFilename && optFilename && *optFilename) {
-        return exportImg(*optFilename, imgFilename);
-    }
-
-    if (showVersion) {
-        g_printf("%s %s \n", PROJECT_NAME, PROJECT_VERSION);
-        g_printf("└──%s: %d.%d.%d \n", "libgtk", gtk_get_major_version(), gtk_get_minor_version(),
-                 gtk_get_micro_version());
-        return 0;
-    }
-
-    // Checks for input method compatibility
-
+    std::unique_ptr<GladeSearchpath> gladePath;
+    std::unique_ptr<Control> control;
+    std::unique_ptr<MainWindow> win;
```

Comment:
These don't need to be pointers.

Reply:
The problem is, that they must initialized delayed and the pointers must persist across the signals.